CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+# Contributing
+
+When contributing to this repository as a curriculum developer, please first
+discuss the change you wish to make via issue, email, or any other method with
+James Cranston before create a feature request branch. 
+
+Please note we have a code of conduct, please follow it in all your
+interactions with the project.
+
+## Merge Request Process for Curriculum Contributions
+
+1. Please create a feature branch for your update in the following format:
+    ```
+    <github username>/contributing/<short description>
+    ```
+    As an example:
+    ```
+    jcranston/contributing/update-project-documentation
+    ```
+   Ensuring branches are named in this format will ensure they do not get
+   picked up by the CI pipeline which is used to test student feature branches
+   who are completing assignments.
+1. Ensure any install or build dependencies are removed before the end of the
+layer when doing a build.
+1. When applicable, update the README.md with details of changes to the
+interface, this includes new environment variables, exposed ports, useful file
+locations and container parameters. Please also update all relevant
+documentation under `docs`. Merge requests without supplemental documentation
+or documentation changes will be rejected.
+1. Increase the version numbers in any examples files and the README.md to the
+new version that this Pull Request would represent. The versioning scheme we
+use is [SemVer](http://semver.org/).
+1. You may merge the Pull Request in once you have the sign off from James
+Cranston through formal code review. If you do not have permission to do that,
+you may request the second reviewer to merge it for you.
+
+## Participating as a Student
+
+Please see [getting-started.md](./docs/getting-started.md) if you are a student
+and would like to attempt any of the assignments in Zero-to-AI. These
+instructions detail the steps you need to take to check out a feature branch
+to attempt any of the assignments, as well as any environment setup questions
+you might have. Please file a Github
+[issue](https://github.com/jcranston/zero-to-ai/issues) if you need help or if
```

Comment:
put links at the bottom of the file and refer to them as variables (see example at line 116 and 118)